pkg/apiserver/apis/v1alpha3/ipblockusage_types.go

```diff
@@ -0,0 +1,68 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	GroupName    = "nsx.vmware.com"
+	Version      = "v1alpha3"
```

Comment:
Why it should be v1alpha3?

Reply:
I do not think this version trick is the right approach, let us use a different APIGroup. I am thinking about the following:
1. Use an APIGroup like vpc.nsx.vmware.com or a name that can indicate it is aggregated API not CRD.
2. If we believe we will finally remove or rename CRD APIGroup in future, e.g. to crd.nsx.vmware.com, then we can go your way of using nsx.vmware.com for API aggregation and use a new version.

What you guys think? @dantingl @lxiaopei

Reply:
"apiservice.nsx.vmware.com" can be one way.
Or we change CRD group to "crd.nsx.vmware.com" in future, and use "nsx.vmware.com" or "vpc.nsx.vmware.com" for APIService resources.
Let us see if @tnqn has an opinion.

Reply:
You may also consider using "extensions.nsx.vmware.com" to hold the new API, using "extensions" to represent this is served by extension apiserver.
Later if it's decided CRDs should go to "crd.nsx.vmware.com", this API may be kept in this group or be moved to "nsx.vmware.com".

Reply:
Do you still prefer not to add "extensions" to APIService group, if we start from scratch with no existing group? If so, another choice is to use "vpc.nsx.vmware.com".

Reply:
Yes, it's good to move the existing CRD  group nsx.vmware.com to crd.nsx.vmware.com. We use extensions.nsx.vmware.com for aggregated APIService group. Does it make sense?

Reply:
For me it depends on whether the API is supposed to do currently and in the future.

From other threads, it seems the API's backend API is "/policy/api/v1/orgs/default/projects/project-1/vpcs/vpc-1/ip-address-usage", which seems to get all usages of all IPBlocks used by this VPC. I assume the API will not evolve to represent IPBlocks themselves, then it's specific to VPC and "vpc.nsx.vmware.com" would make sense to me. (BTW, why the API doesn't follow the API name to be "IPAddressUsage"?)

But if it's the opposite, i.e., it's supposed to CRUD IPBlocks in the future, then it seems not proper to be specific to VPC?

Reply:
If my understanding it correct, I think "IPBlockUsage" may be a bit confusing as it sounds the usage of an IPBlock and user may expect the resource name to be IPBlock, but it's actually about the IP usages of a single VPC on multiple IPBlocks, the resource name would be VPC name, right? If yes, I feel the API name should be VPCIPAddressUsage, or IPAddressUsage since it's already in VPC API group.

Reply:
Thanks for Quan. This API is designed to show IpAddress usage inside one namesapce. Of course, current, there is only VPC created for one VPC. so, vpc.nsx.vmware.com would make more sense. 

Also, IPAddressUsage name is more appropriate to align with NSX API side, Thanks for advice.

Reply:
Sure, Let's use vpc.nsx.vmware.com as for now.

Reply:
vpc.nsx.vmware.com makes sense to me.